mmedit/datasets/sr_folder_ref_dataset.py

```diff
@@ -0,0 +1,120 @@
+import os.path as osp
+
+from .base_sr_dataset import BaseSRDataset
+from .registry import DATASETS
+
+
+@DATASETS.register_module()
+class SRFolderRefDataset(BaseSRDataset):
+    """
+    General paired image folder dataset for reference-based image restoration.
+
+    The dataset loads ref (reference) image pairs
+        Must contain: ref (reference)
+        Optional: GT (Ground-Truth), LQ (Low Quality), or both
```

Comment:
Is there any case where only reference is needed? I think at least GT should be required?

Reply:
One of the follows:
- Ref and LQ
- Ref and GT,
- Ref, LQ, and GT,

Reply:
Would it be better to raise error if `lq_folder` and `gt_folder` are both `None`?